### ISSUE TITLE
a11y checkup

### ISSUE THREAD

**Issue by gregrickaby** [WebDevStudios/wd_s#276]
WDS is committed to building accessible websites. Continually run the theme through both Total Validator app and the WAVE extension for Chrome.

- [ ] Section 508
- [ ] WCAG 2.0AA
- [ ] HTML Validation
- [ ] CSS Validation

**Comment by Cheffheid** [comment]
I've pushed a handful of fixes for T.V. and will continue to work on the same branch for things I'm finding.

The branch is hotfix/a11y.

What I've fixed/changed on that branch so far:

- Removed `role="main"` from the main elements (T.V. pleaser, it flags `<nav role="navigation">` and `<aside role="complementary">` as warnings, but they're essentially the same thing...)
- Set widget title headings to h2 (up from h3), like in _s and twentyseventeen. And added some styles that let the `.widget-title` class inherit the .h3 class so that it won't just break (not set by default though, so a default install is still going to have larger widget titles).
- Added a hidden heading to the comment section for the sake of separation.
- Removed the "cancel reply" link from the "Leave a reply" heading. It's just confusing, and it doesn't even really seem to work (it's hidden by default, but I've not been able to unhide it). Ideally, Core is updated to move it out of the heading. I'll investigate that more on a 5ftf, because that's may be a nightmare style-wise.

This should take care of most of the errors that T.V. produces. Most, because it also produces one error on the url field in the comment form that's really baffling:

![wat](https://cloud.githubusercontent.com/assets/1461585/26798705/9bc592e8-49ff-11e7-9524-d9b26fc84380.png)

On the URL field in the comment form, it decides to flag an error with the message "The 'value' attribute does not have a valid value: It must be an absolute URL". Even if this wasn't a field produced by core, there is no way to fix this and I don't think it's even a valid error from a markup perspective.

**Comment by Cheffheid** [comment]
Added some additional fixes on that branch that automatic testing won't necessarily pick up. They are related to the submenus and colour contrast.

The submenus weren't displaying on focus, and now they are. While I was at it, I've also added the caret down icon to the menu links that have a submenu. I did notice that the submenus have no background set (not sure if that's intentional or not) and I've left that the way it is for now which is what you're seeing in the GIF below.

![navigation](https://user-images.githubusercontent.com/1461585/26949296-6bc833c4-4c67-11e7-81c0-521569f98272.gif)

For the comments, the timestamp on them was darkened and the "$color-blue" variable was darkened a little as well. A before and after below (before on the left, after on the right).

![comment-contrast](https://user-images.githubusercontent.com/1461585/26949347-9e1c6214-4c67-11e7-8102-afb3ab8c2a8a.png)

**Comment by coreymcollins** [comment]
We have some major markup updates coming in #447, and I know @mharis has already worked on bringing the dropdown menus up to speed a11y-wise.

I'm going to close this issue out for now and open a new one after @jomurgel's grid branch is merged in, as we'll have a bunch of new markup on our hands that wasn't there when this issue was created back in 2017. Plus, most of the HTML and CSS validation should be in a much better place (and already done!) with the way we've worked linting into our processes.
